diff --git a/Makefile b/Makefile
index 0b2e57a..cd09e04 100644
--- a/Makefile
+++ b/Makefile
@@ -61,6 +61,8 @@ TARGET_DTS_FILES:=$(foreach dts,$(TARGET_DTS_FILES),build/$(dts))
 TOOLCHAIN:
 	make -C buildroot ARCH=arm zynq_$(TARGET)_defconfig
 	make -C buildroot toolchain
+	( cd buildroot/output/host/bin; for i in *; do newname="$$(echo "$$i"|sed s/arm-none-linux-gnueabihf-/arm-linux-gnueabihf-/)"; test -L "$$newname" && rm "$$newname"; ln -s "$$(realpath "$$i")" "$$newname" || true; done )
+	( cd buildroot/output/host/opt/ext-toolchain/bin; for i in *; do newname="$$(echo "$$i"|sed s/arm-none-linux-gnueabihf-/arm-linux-gnueabihf-/)"; test -L "$$newname" && rm "$$newname"; ln -s "$$(realpath "$$i")" "$$newname" || true; done )
 
 build:
 	mkdir -p $@
@@ -90,6 +90,7 @@ build/uboot-env.bin: build/uboot-env.txt
 
 linux/arch/arm/boot/zImage: TOOLCHAIN
 	$(TOOLS_PATH) make -C linux ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) zynq_$(TARGET)_defconfig
+	$(TOOLS_PATH) make -C linux ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) menuconfig
 	$(TOOLS_PATH) make -C linux -j $(NCORES) ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) zImage UIMAGE_LOADADDR=0x8000
 
 
diff --git a/scripts/pluto.its b/scripts/pluto.its
index e69e417..aa7fa08 100644
--- a/scripts/pluto.its
+++ b/scripts/pluto.its
@@ -156,8 +156,8 @@
 		};
 
 		config@9 { /* This one is actually RevB */
-			description = "Linux with fpga RevB";
-			fdt = "fdt@2";
+			description = "Linux with fpga RevC";
+			fdt = "fdt@3";
 			kernel = "linux_kernel@1";
 			ramdisk = "ramdisk@1";
 			fpga = "fpga@1";
